src/hooks/use-stores.ts

```diff
@@ -57,6 +57,19 @@ export function useDocumentOrCurriculumMetadata(key?: string): IDocumentMetadata
   }, [documentMetadata, key]);
 }
 
+export function useTypeOfTileInDocumentOrCurriculum(key?: string, tileId?: string) {
+  const { documents } = useStores();
+  if (!key || !tileId) return;
```

Comment:
Would we even hit this case (no key or no tile Id)?

Reply:
Note that we also have an unknown tool type (`export const kUnknownToolID = "Unknown";`) so it might be fair to simply return that.

Reply:
>I'm really not a fan of having functions returning undefined! Is there a reasonable use-case for that behavior? If not, I'd prefer that we throw an error here.

Throwing an exception is a fairly extreme response to a request for information that doesn't exist for some reason. Returning `undefined` is a common way of indicating that requested information doesn't exist, e.g. `Array.find()`.

>Would we even hit this case (no key or no tile Id)?

Also note that this is implemented as a hook. Hook functions can't be called conditionally, so you can't just say that clients should check the arguments themselves like this:
```typescript
if (documentKey && tileID) {
  tileType = useTypeOfTileInDocumentOrCurriculum(documentKey, tileID);
  ...
}
```
because that would violate the rules of hooks.

In the specific context of commenting, sometimes no document is selected and sometimes a document is selected but no tile is selected. The hook is designed to be callable in all situations and to do something reasonable.

Reply:
Seems reasonable to me.

Reply:
>Note that we also have an unknown tool type (`export const kUnknownToolID = "Unknown";`) so it might be fair to simply return that.

I failed to respond to one part of your comment. The `kUnknownToolID` was something I put in place at the very beginning to solve a very specific problem that I've encountered in other component-based applications and that we haven't yet run into in CLUE but will shortly if we continue down the path of supporting plugin tiles. When reading in tile-based content you have to be able to deal with the situation in which the document contains tile content from tiles that are not currently available. In this case, we can instantiate the Unknown tool which has the crucial property that it retains the serialized content and will re-serialize it on write. As a result, you should be able to open up a document in which a tile is not recognized and then re-save the document and open it later in a context in which the tile content is recognized and the content will have been preserved. As such, it's not an appropriate response to a request that's missing a document key or tile id.